aws/resource_aws_sns_topic_subscription.go

```diff
@@ -101,11 +101,24 @@ func resourceAwsSnsTopicSubscriptionCreate(d *schema.ResourceData, meta interfac
 	return resourceAwsSnsTopicSubscriptionUpdate(d, meta)
 }
 
+func wasSetAndHasChange(d *schema.ResourceData, key string) bool {
+	if d.HasChange(key) {
+		original, _ := d.GetChange(key)
+		if original.(string) == "" {
+			return false
+		}
+		return true
+	} else {
+		return false
+	}
+}
+
 func resourceAwsSnsTopicSubscriptionUpdate(d *schema.ResourceData, meta interface{}) error {
 	snsconn := meta.(*AWSClient).snsconn
 
-	// If any changes happened, un-subscribe and re-subscribe
-	if d.HasChange("protocol") || d.HasChange("endpoint") || d.HasChange("topic_arn") {
+	// If any changes happened, un-subscribe and re-subscribe.
+	// We also have to handle the "freshly-created subscription" case, where all of the fields have "changed", but only from their original value of empty strings ("").
+	if wasSetAndHasChange(d, "protocol") || wasSetAndHasChange(d, "endpoint") || wasSetAndHasChange(d, "topic_arn") {
```

Comment:
I think we can avoid adding the wrapper function and use `d.IsNewResource()` instead here.

e.g.

```go
if !d.IsNewResource() && (d.HasChange("protocol") || d.HasChange("endpoint") || d.HasChange("topic_arn")) {
```